### DISCUSSION TITLE
Encoding of the email addresses

### DISCUSSION THREAD

**Discussion by sandissauka** [craftcms/cms#12300] [Ideas] [answered]
There's [Sprout Encode Email](https://plugins.craftcms.com/sprout-encode-email?craft3) that is not supported anymore. The plugin encodes "the email addresses in your templates so they can't be harvested by evil spam bots". Maybe it could be integrated into Craft (including 3.x)?

**UPDATE 14-NOV-2022:**
The issue in addition is that this plugin doesn't set cookies. So if a user chooses only necessary cookies, e-mails are hidden. There's no way put the cookie in the category "Necessary".

P.S. Started from [this discussion](https://github.com/barrelstrength/craft-sprout-encode-email/issues/8).

**Comment by thupsi** [comment]
There is also this one: https://plugins.craftcms.com/obfuscator?craft4

**Comment by lukas-jansen** [comment]
If you have your domain in Cloudflare you could also their email encoding, it will automatically encode mailto links.

**Comment by sandissauka** [comment]
**UPDATE 14-NOV-2022:** (added to the description)
The issue in addition is that this plugin doesn't set cookies. So if a user chooses only necessary cookies, e-mails are hidden. There's no way put the cookie in the category "Necessary".


Both suggestions might lead to the same issue.

**Comment by brandonkelly** [comment, marked as answer]
We discussed this internally and don’t feel like it would be a good idea for Craft to do this out of the box, as there are email harvesting bots that are known to support encoded emails. If your email infrastructure isn’t ready to handle spam, you probably shouldn’t be including email addresses on your website, encoded or not.
